homeassistant/components/climate/evohome.py

```diff
@@ -0,0 +1,47 @@
+"""Support for Honeywell evohome (EMEA/EU-based systems only).
+
+Support for a temperature control system (TCS, controller) with 0+ heating
+zones (e.g. TRVs, relays) and, optionally, a DHW controller.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/evohome/
+"""
+
+import logging
+
+from homeassistant.components.evohome import (
+    EvoController,
+
+    DATA_EVOHOME,
+    CONF_LOCATION_IDX,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create a Honeywell (EMEA/EU) evohome CH/DHW system.
+
+    An evohome system consists of: a controller, with 0-12 heating zones (e.g.
+    TRVs, relays) and, optionally, a DHW controller (a HW boiler).
+
+    Here, we add the controller, and the zones (if there are any).
+    """
+    client = hass.data[DATA_EVOHOME]['client']
+    loc_idx = hass.data[DATA_EVOHOME]['params'][CONF_LOCATION_IDX]
+
+# Collect the (master) controller - evohomeclient has no defined way of
```

Comment:
Although not necessary for this PR, you should try to make this data available in evohomeclient without requiring protected access.

Reply:
The code owner has previously been reluctant to go for multiple locations.

I'll have a chat with him about it, he's accepted few PRs off me so by presenting him with another PR he may well go for it.